### PR TITLE
docs(sdk): add documentation to Object3D for individual methods

### PR DESCRIPTION
**Pull request by ssisk**
Fixes WB-12027

Description
-----------
We were missing docstrings for individual parameters of Object3D - this adds those.

Testing
-------
Docs only

Checklist
-------
- [x] Include reference to internal ticket "Fixes WB-NNNN" and/or GitHub issue "Fixes #NNNN" (if applicable)
- [x] Ensure PR title compliance with the [conventional commits standards](https://github.com/wandb/wandb/blob/main/CONTRIBUTING.md#conventional-commits)

### CODE REVIEW

File: wandb/sdk/data_types/object_3d.py [hunk 1]
```diff
@@ -233,6 +227,20 @@ def from_file(
...
+        Arguments:
+            data: (numpy array): Each entry in the array will
+                represent one point in the point cloud.
+
+
+        The shape of the numpy array must be one of either:
+        ```
+        [[x y z],       ...] nx3
+        [[x y z c],     ...] nx4 where c is a category with supported range [1, 14]
+        [[x y z r g b], ...] nx4 where is rgb is color
```
Comment (dmitryduev) [inline wandb/sdk/data_types/object_3d.py:242]:
```suggestion
        [[x y z r g b], ...] nx4 where is rgb is color
        ```
```

**Review by dmitryduev — APPROVED** [review]

File: wandb/sdk/data_types/object_3d.py [hunk 2]
```diff
@@ -255,6 +264,17 @@ def from_point_cloud(
         point_cloud_type: "PointCloudType" = "lidar/beta",
         # camera: Optional[Camera] = None,
     ) -> "Object3D":
+        """
+        Initializes Object3D from a python object.
+
+        Arguments:
+            points (Sequence["Point"]): The points in the point cloud.
+            boxes (Sequence["Box3D"]): 3D bounding boxes intended to allow labeling parts of the point cloud. These
```
Comment (ngrayluna) [inline wandb/sdk/data_types/object_3d.py:272]: Change to active voice:

boxes (Sequence["Box3D"]): 3D bounding boxes for labeling the point cloud. Boxes are displayed in point cloud visualizations.

File: wandb/sdk/data_types/object_3d.py [hunk 3]
```diff
@@ -233,6 +227,21 @@ def from_file(
 
     @classmethod
     def from_numpy(cls, data: "np.ndarray") -> "Object3D":
+        """
+        Initializes Object3D from a numpy array.
+
+        Arguments:
+            data: (numpy array): Each entry in the array will
```
Comment (ngrayluna) [inline wandb/sdk/data_types/object_3d.py:234]: Nit:  extra colon after `data` 

Arguments:
     data (numpy array): .....

File: wandb/sdk/data_types/object_3d.py [hunk 4]
```diff
@@ -225,6 +211,14 @@ def from_file(
         data_or_path: Union["TextIO", str],
         file_type: Optional["FileFormat3D"] = None,
     ) -> "Object3D":
+        """
+        Initializes Object3D from a file or stream
+
+        Arguments:
+            data_or_path (Union["TextIO", str]): A path to a file or a `TextIO` stream.
+            file_type (str): Specifies the format of data in `data_or_path`. Required when `data_or_path` is a
+                stream, ignored when `data_or_path` is a path (the type will be taken from the file extension.)
```
Comment (ngrayluna) [inline wandb/sdk/data_types/object_3d.py:220]: Specifies the data format passed to `data_or_path`. Required when `data_or_path` is a `TextIO` stream.  This parameter is ignored if a file path is provided. The type is taken from the file extension.

File: wandb/sdk/data_types/object_3d.py [hunk 5]
```diff
@@ -255,6 +264,17 @@ def from_point_cloud(
...
     ) -> "Object3D":
+        """
+        Initializes Object3D from a python object.
+
+        Arguments:
+            points (Sequence["Point"]): The points in the point cloud.
+            boxes (Sequence["Box3D"]): 3D bounding boxes intended to allow labeling parts of the point cloud. These
+                will be displayed in the point cloud visualization.
+            vectors (Optional[Sequence["Vector3D"]]): Each vector will be displayed in the point cloud
+                visualization. Can be used to indicate directionality of bounding boxes. Defaults to None.
```
Comment (ngrayluna) [inline wandb/sdk/data_types/object_3d.py:275]: Nit: change verbage to active voice

Vectors (Optional[Sequence["Vector3D"]]): Each vector is displayed in the point cloud
                visualization. Can be used to indicate directionality of bounding boxes. Defaults to `None`.

File: wandb/sdk/data_types/object_3d.py [hunk 6]
```diff
@@ -255,6 +264,17 @@ def from_point_cloud(
         point_cloud_type: "PointCloudType" = "lidar/beta",
         # camera: Optional[Camera] = None,
     ) -> "Object3D":
```
Comment (ngrayluna) [inline wandb/sdk/data_types/object_3d.py:266]: It was initially unclear what the types where for points, boxes, and vectors for `from_point_cloud` (e.g. `Sequence['Points']`).

Would it be improper to instead to "array" or list of dictionaries, etc.?

**Review by ngrayluna — COMMENTED** [review]
Descriptions made sense. Word-smithed a couple of them to make it in the active voice tense.

Only thing that threw me a bit off where the type descriptions for `from_point_cloud`. Looking at the API by itself it wasn't entirely clear what `Sequence['Points']` was.   Goggled and found info on the `typing` module so that cleared up Sequence.

Don't know if folks are fairly familiar w/ that format and library. If yay, I'd say keep as is. If not,maybe change it from:

`points (Sequence["Point"])` to  `points (Sequence)`

Or say it is an `array` or `list of dict`?

File: wandb/sdk/data_types/object_3d.py [hunk 7]
```diff
@@ -198,10 +198,12 @@ def __init__(
 
             if len(np_data.shape) != 2 or np_data.shape[1] not in {3, 4, 6}:
                 raise ValueError(
-                    """The shape of the numpy array must be one of either
+                    """
+                    The shape of the numpy array must be one of either
                                     [[x y z],       ...] nx3
                                      [x y z c],     ...] nx4 where c is a category with supported range [1, 14]
-                                     [x y z r g b], ...] nx4 where is rgb is color"""
+                                     [x y z r g b], ...] nx4 where rgb is color
```
Comment (kptkin) [inline wandb/sdk/data_types/object_3d.py:205]: is it actually nx4 should it be nx6?

Also it seems misleading that we have different length in the same array?

**Review by dmitryduev — APPROVED** [review]
LGTM! Thanks for the effort!

File: wandb/sdk/data_types/object_3d.py [hunk 8]
```diff
@@ -232,15 +241,31 @@ def from_file(
...
+        Arguments:
+            data (numpy array): Each entry in the array will
+                represent one point in the point cloud.
+
+
+        The shape of the numpy array must be one of either:
+        ```
+        [[x y z],       ...] nx3
+        [[x y z c],     ...] nx4 where c is a category with supported range [1, 14]
+        [[x y z r g b], ...] nx6 where is rgb is color
```
Comment (moredatarequired) [inline wandb/sdk/data_types/object_3d.py:255]:
```suggestion
        [[x y z],       ...]  # nx3.
        [[x y z c],     ...]  # nx4 where c is a category with supported range [1, 14].
        [[x y z r g b], ...]  # nx6 where is rgb is color.
```

**Review by moredatarequired — APPROVED** [review]

File: wandb/sdk/data_types/object_3d.py (hunk 8 above)
Comment (ssisk) [reply wandb/sdk/data_types/object_3d.py:255]: For my own learning: what's the purpose of the # ? As someone mostly unfamiliar with python docs I don't know the convention around that.

**Pull request merged**